crates/sui-json-rpc/src/api.rs

```diff
@@ -116,6 +117,15 @@ pub trait RpcReadApi {
 #[open_rpc(namespace = "sui", tag = "Full Node API")]
 #[rpc(server, client, namespace = "sui")]
 pub trait RpcFullNodeReadApi {
+    #[method(name = "callTransaction")]
+    async fn call_transaction(
+        &self,
+        tx_bytes: Base64,
+        sig_scheme: SignatureScheme,
+        signature: Base64,
+        pub_key: Base64,
+    ) -> RpcResult<SuiTransactionEffects>;
+
```

Comment:
just my personal opinion :) 

will calling it `dry_run_transaction` makes it more obvious that this method doesn't commit? call_transaction is a bit ambitious for me.

Reply:
also I think it makes more sense to put it in `QuorumDriverApi` together with `execute_transaction`

Reply:
@patrickkuo  `dry_run_transaction` is a better name indeed, will change it accordingly!

> to put it in QuorumDriverApi 
I was thinking about this as well, if I do it this way, iirc the "dry_tx" will go to `NetworkAuthorityClient` and then try to reach validators, instead of executing it on the local fullnode following the current impl.

Executing locally on the fullnode is better I think bc 
- its not necessary to involve validators here
- dry run request traffic on the fullnode is isolated from other nodes in case of abusing

Did I miss sth?

Reply:
to do this we will need to pass in the AuthorityState to the FullNodeQuorumDriverApi, this should execute locally.

Having 2nd thoughts, the QuorumDriver is configurable, so fullnode can choose to disable it, maybe it's not a good idea to put it in QuorumDriverApi depends on the use case.

Reply:
> the QuorumDriver is configurable, so fullnode can choose to disable it

right, iirc QuorumDriver drives txns with a quorum of validators as the name says, thus dry_run to me is more like a read-only endpoint.